plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -412,3 +429,15 @@ func MapPodsToMachines(lister algorithm.PodLister) (map[string][]*api.Pod, error
 	}
 	return machineToPods, nil
 }
+
+// search two arrays and return true if they have at least one common element; return false otherwise
+func haveSame(a1, a2 []string) bool {
```

Comment:
Not the most efficient implementation of disjoint set check. But, since the arrays are tiny, nbd.